src/GraphQL/Types/Scalars/EnumerationGraphType.cs

```diff
@@ -128,9 +128,14 @@ public EnumValueDefinition FindByName(string name, StringComparison comparison =
 
         public EnumValueDefinition FindByValue(object value)
         {
+            if (value is Enum)
+            {
+                value = Convert.ChangeType(value, Enum.GetUnderlyingType(value.GetType()));
+            }
+
             // DO NOT USE LINQ ON HOT PATH
             foreach (var def in _values)
-                if (def.Value.Equals(value))
+                if (def.UnderlyingValue.Equals(value))
```

Comment:
By the way, think about such situation:
```c#
enum ParityEnum
{
  Zero = 0,
  One = 1,
  Two = 0,
  Three = 1,
  Four = 0
}
```

CLR enum itself can be thought of as a pair of name and arbitrary (not necessarily unique) value. You probably guess what strange errors can be in this case, if you start comparing by `UnderlyingValue` instead of `Value`? The results will vary depending on the order in which the enumeration items are registered.

Maybe we can eleminate these cases by that code?
```c#
 public EnumValueDefinition FindByValue(object value)
        {
            // DO NOT USE LINQ ON HOT PATH
            foreach (var def in _values)
                if (def.Value.Equals(value))
                    return def;

            if (value is Enum)
            {
                value = Convert.ChangeType(value, Enum.GetUnderlyingType(value.GetType()));

                // DO NOT USE LINQ ON HOT PATH
                foreach (var def in _values)
                    if (def.UnderlyingValue.Equals(value))
                        return def;
            }

            return null;
        }
```

Reply:
Good idea, but doesn't work in practice.  C# equates enumeration members with the same value.  So `ParityEnum.Zero == PairtyEnum.Two` returns `true`.

Reply:
It is also noteworthy that `Console.WriteLine(ParityEnum.Zero)` prints the text `Four`, so name based matching would not work either.

Reply:
Yeah. I can guess why. Enumeration of values goes from the end. There are always some problems with enums.